Add documentation for SKR tests (#10)
kyma-project/kyma-environment-broker: docs/contributor/05-20-e2e_tests.md

> # E2E tests of kyma-environment-broker

Author: I wanted to keep it consistent with the [btp-manager](https://github.com/kyma-project/btp-manager/blob/main/docs/contributor/05-20-e2e_tests.md#e2e-tests-of-btp-manager-installation-from-oci-module-image)

> # E2E tests of kyma-environment-broker

Reviewer: Let's use Kyma Environment Broker. I'll adjust the BTP Manager doc.

## Overview

The end-to-end (E2E) tests covers kyma-environment-broker (KEB) and SAP BTP, Kyma runtime (SKR).
There are four tests:

> - `skr-tests` for checking on different cloud service providers Kyma provisioning, BTP Manager secret reconcilation, updating OIDC, updating machine type and Kyma deprovisioning

Author: What about this:

> - `skr-tests` for testing the following operations on different cloud service providers: Kyma provisioning, BTP Manager secret reconcilation, updating OIDC, updating machine type and Kyma deprovisioning

> - `skr-tests` for checking on different cloud service providers Kyma provisioning, BTP Manager secret reconcilation, updating OIDC, updating machine type and Kyma deprovisioning

Reviewer: Much better :)

> - `skr-tests` for testing the following operations on different cloud service providers: Kyma provisioning, BTP Manager secret reconcilation, updating OIDC, updating machine type and Kyma deprovisioning

Reviewer: ```suggestion
- `skr-tests` for testing the following operations on different cloud service providers: Kyma provisioning, BTP Manager secret reconciliation, updating OIDC, updating machine type, and Kyma deprovisioning
```

#### Without provisioning

In this mode the test additionally needs the following environment variables:
- **SKIP_PROVISIONING** set to `true`
- **INSTANCE_ID** the UUID of the provisioned SKR instance

In this mode, the test executes the following steps:
1. Ensures the SKR exists.
2. Runs OIDC test.
3. Cleans up the resources.
 
### Test execution

1. Before you run the test, prepare the `.env` file based on the following [.env.template](/testing/e2e/skr/skr-test/.env.template):
2. To set up the environment variables in your system, run:

```bash
export $(xargs < .env)
```

3. Choose whether you want to run the test with or without provisioning.
    - To run the test with provisioning, call the following target:

    ```bash
    make skr
    ```
    - To run the SKR test without provisioning, use the following command:

    ```bash
    make skr SKIP_PROVISIONING=true
    ```

## E2E SKR own cluster test

### Usage

The test executes the following steps:
1. Provisions SKR on a cluster given by the customer
2. Checks if sap-btp-manager secret exist.
3. Deprovisions SKR on a cluster given by the customer

### Test execution 

> 1. Before you run the test, prepare the `.env` file based on the following [.env.template](/testing/e2e/skr/skr-test/.env.template):

Reviewer: ```suggestion
1. Before you run the test, prepare the `.env` file based on the following [`.env.template`](/testing/e2e/skr/skr-test/.env.template):
```

2. To set up the environment variables in your system, run:

```bash
export $(xargs < .env)
```

3. Run the test scenario:
```bash
make skr-own-cluster
```

## E2E SKR AWS upgrade integration test

### Usage

The test executes the following steps:
1. Provisions an SKR cluster.
2. Runs Kyma upgrade.

> 3. Deprovisions the SKR instance and clean up the resources.

Reviewer: ```suggestion
3. Deprovisions the SKR instance and cleans up the resources.
```

### Test execution 

> 1. Before you run the test, prepare the `.env` file based on the following [.env.template](/testing/e2e/skr/skr-aws-upgrade-integration/.env.template):

Reviewer: ```suggestion
1. Before you run the test, prepare the `.env` file based on the following [`.env.template`](/testing/e2e/skr/skr-aws-upgrade-integration/.env.template):
```

2. To set up the environment variables in your system, run:

```bash
export $(xargs < .env)
```

3. Run the test scenario:
```bash
make skr-aws-upgrade-integration
```

## KEB endpoints test

### Usage

The test executes the following steps:
1. Calls KEB endpoints without an authorization token.
2. Checks whether the call was rejected.

### Test execution 

> 1. Before you run the test, prepare the `.env` file based on the following [.env.template](/testing/e2e/skr/keb-endpoints-test/.env.template):

Reviewer: ```suggestion
1. Before you run the test, prepare the `.env` file based on the following [`.env.template`](/testing/e2e/skr/keb-endpoints-test/.env.template):
```

2. To set up the environment variables in your system, run:

```bash
export $(xargs < .env)
```

3. Run the test scenario.
```bash
make keb-endpoints
```

### CI pipelines

> The tests are run once per day at 01:05 by the given prowjobs:

Reviewer: ```suggestion
The tests are run once per day at 01:05 by the given Prow jobs:
```
